### PR TITLE
[IMPROVEMENT] Migrate subscriptions & rooms fetching to HTTP

### LINKED ISSUE #1517: [IMPROVEMENT] Move "rooms list" & "subscriptions list" call from WebSocket to REST
API to get rooms: https://rocket.chat/docs/developer-guides/rest-api/rooms/get/
API to get subscriptions: https://rocket.chat/docs/developer-guides/rest-api/subscriptions/get/

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

Closes #1517

### CODE REVIEW

**Comment by cardoso** [comment]
Tests seem to be failing randomly. Trying to fix them.

**Comment by cardoso** [comment]
Tests are ok now @rafaelks ✅

File: Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift [hunk 1]
```diff
@@ -22,121 +22,16 @@ struct SubscriptionManager {
...
-                        }
-                    }
-
-                    realm.add(subscriptions, update: true)
-                }, completion: {
-                    completion(response)
-                })
+        client?.fetchSubscriptions(updatedSince: lastUpdate) {
+            client?.fetchRooms(updatedSince: lastUpdate) {
+                completion?()
```
Comment (rafaelks) [inline Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift]: The only issue I see here is: if we've any error we're not continuing the app execution, right? Only if the error is the fallback version, of course. But what if by any reason one request fails with a 400 or 500? I think we can find a way to continue the execution here.

What do you think @cardoso?

**Comment by rafaelks** [comment]
@cardoso This is looking great, I just made one comment. Let's discuss what's the best solution. 👍

File: Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift (hunk 1 above)
Comment (cardoso) [reply Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift]: I can think of a few options

1. Try again after some time (5s?)
2. Disconnect the user from the server
3. Continue execution despite outdated subscriptions/rooms

I'm torn between trying again and disconnecting the user from the server

Comment (rafaelks) [reply Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift]: @cardoso As we spoke:

1. Let's retry 3 times after 3 seconds on failure; After that, just ignore the error;
2. Always continue the flow, never logout the user;

File: Rocket.Chat/API/API.swift [hunk 2]
```diff
@@ -41,6 +48,8 @@ final class API: APIFetcher {
     var authToken: String?
     var userId: String?
 
+    var retriesOnError: Int = 0
```
Comment (rafaelks) [inline Rocket.Chat/API/API.swift]: This is unused?

Comment (cardoso) [reply Rocket.Chat/API/API.swift]: Nice catch :+1:

**Pull request merged**